metabase/metabase
# Separate SMTP Connection Setup from Other Email Settings (no redirects version)

**Problem:** Half of the email settings, the SMTP connection configuration, is **click-to-save**, while the other half, the additional email settings, is **auto-save**. This is a confusing paradigm for users.

**Context:**
- https://github.com/metabase/metabase/issues/32818

---
**Desired Behavior**/**Acceptance Criteria**

_Behaviors_
- [ ] SMTP connection configuration moved into separate setup placed behind a configuration card
  - [ ] Breadcrumbs have been added for back-navigation to top-level email settings
- [ ] All other functionality behaves as originally intended (saving, sending test emails, clearing the form, etc.)

_Tests_
- [ ] Tests updated to configure the SMTP connection through the configuration card
- [ ] All other behaviors work as intended (saving settings, sending test emails, clearing settings, etc.)

---
**How to Verify**

https://github.com/metabase/metabase/assets/22608765/0da46f22-d9c6-46a7-8c02-89f6af120a39

1. Log in as admin
2. (if SMTP is already setup) Clear SMTP configuration
3. Go to admin settings > email (from the sidebar)
4. See that the SMTP configuration is now placed behind a configuration card
5. Click set up
6. Setup a valid SMTP configuration
7. Navigate back to top-level email settings (using breadcrumbs)
8. See that the configuration card changed to `Active` and the button is now `Edit Configuration`
9. Test that the additional settings are configurable and auto-save
10. Test that already other behaviors work as intended (saving, clearing, sending test emails, etc.)
